mod/wikirate_source/spec/set/type/source/source_spec.rb

```diff
@@ -61,182 +85,184 @@
-      sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :file=>test_csv,:type_id=>Card::FileID}}
+      sourcepage = create_source file: test_csv
       html = sourcepage.format.render_metric_import_link
-      source_file = sourcepage.fetch :trait=>:file
-      expect(html).to have_tag("a",:with=>{:href=>"/#{source_file.cardname.url_key}?view=import"},:text=>"Import to metric values")
+      source_file = sourcepage.fetch trait: :file
+      expect(html).to have_tag("a",with: {href: "/#{source_file.cardname.url_key}?view=import"},text: 'Import to metric values')
     end
-    describe "original_icon_link" do
-      context "file source" do
-        it "renders upload icon" do
+    describe 'original_icon_link' do
+      context 'file source' do
+        it 'renders upload icon' do
           test_csv = File.open("#{Rails.root}/mod/wikirate_source/spec/set/type_plus_right/source/import_test.csv")
-          sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :file=>test_csv,:type_id=>Card::FileID}}
+          sourcepage = create_source file: test_csv
           html = sourcepage.format.render_original_icon_link
-          source_file = sourcepage.fetch :trait=>:file
-          expect(html).to have_tag("a",:with=>{:href=>source_file.file.url}) do
-            with_tag "i",:with=>{:class=>"fa fa-upload"}
+          source_file = sourcepage.fetch trait: :file
+          expect(html).to have_tag("a",with: {href: source_file.file.url}) do
+            with_tag "i",with: {class: 'fa fa-upload'}
           end
         end
       end
-      context "link source" do
-        it "renders globe icon" do
+      context 'link source' do
+        it 'renders globe icon' do
           html = @source_page.format.render_original_icon_link
-          expect(html).to have_tag("a",:with=>{:href=>@url}) do
-            with_tag "i",:with=>{:class=>"fa fa-globe"}
+          expect(html).to have_tag("a",with: {href: @url}) do
+            with_tag "i",with: {class: 'fa fa-globe'}
           end
         end
       end
-      context "text source" do
-        it "renders pencil icon" do
-          new_sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+Text'=>{:type_id=>Card::BasicID,:content=>"test text report"} }
+      context 'text source' do
+        it 'renders pencil icon' do
+          new_sourcepage = create_source text: 'test text report'
           html = new_sourcepage.format.render_original_icon_link
-          text_source = new_sourcepage.fetch(:trait=>:text)
-          expect(html).to have_tag("a",:with=>{:href=>"/#{text_source.cardname.url_key}"}) do
-            with_tag "i",:with=>{:class=>"fa fa-pencil"}
+          text_source = new_sourcepage.fetch(trait: :text)
+          expect(html).to have_tag("a",with: {href: "/#{text_source.cardname.url_key}"}) do
+            with_tag "i",with: {class: 'fa fa-pencil'}
```

Comment:
Space missing after comma.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.